src/Orleans/Streams/Internal/StreamSubscriptionHandleImpl.cs

```diff
@@ -79,14 +86,16 @@ public override Task<StreamSubscriptionHandle<T>> ResumeAsync(IAsyncObserver<T>
                 await NextItem(itemTuple.Item1, itemTuple.Item2);
             }
 
-            expectedToken = StreamHandshakeToken.CreateDeliveyToken(batch.SequenceToken);
-
+            if (isRewindable)
+            {
+                expectedToken = StreamHandshakeToken.CreateDeliveyToken(batch.SequenceToken);
+            }
             return null;
         }
 
         public async Task<StreamHandshakeToken> DeliverItem(object item, StreamSequenceToken currentToken, StreamHandshakeToken handshakeToken)
         {
-            if (expectedToken != null)
+            if (isRewindable && expectedToken != null)
```

Comment:
This line will prevent the subscription handle from sending a correction to the agent telling it where in the cache to read from in cases where data has arrived out of order.  This is probably appropriate for out of order data, but I do not think this is sufficient to resolve the issue.

The subscription handle still retains the expected token, and when the agent is moved, it is this token that becomes the new starting point for delivery, during the handshake process.  Since the data in the cash, in the case of azure queue) will appear to be older than this token, it will not be delivered until the ordering information catches up.